### ISSUE TITLE
Dynamic fields of type checkboxes are not in allowed fields

### ISSUE THREAD

**Issue by staffanm** [jhpyle/docassemble#449]
The following interview results in the error message `The variables 'foo' and "foo['One']" were not in the allowed fields, which were {'foo'}` after answering the first screen (titled `dynamic`):

```yaml
question: static
fields:
  - label: Choose
    field: foo
    datatype: checkboxes
    choices:
      - One
      - Two 
      - Three
      - Four
    none of the above: false
---
question: dynamic
sets: foo
fields:
  - code: myfields
---
code: |
  myfields = [{
    'label': 'Choose',
    'field': 'foo',
    'datatype': 'checkboxes',
    'choices': ['One','Two','Three','Four'],
    'none of the above': False
  }]
---
mandatory: true
question: Ok!
subquestion: |
  You've selected ${foo.true_values()}
```
If the `question: dynamic` block is removed, the equivalent(?) `question: static` block is used instead, which works. The problem does not seem to surface with other datatype fields added by code, only `checkboxes`.

**Comment by jhpyle** [comment]
I believe I have fixed this in 1.2.100.  Sorry it took me so long.  Thanks very much for making it so easy for me to replicate the issue.

**Comment by staffanm** [comment]
Hey, under a week turnaround is amazingly fast! Sorry it took me much longer to reply...

Confirm that this is fixed, closing.
